src/modules/frequency/application/frequency.service.ts

```diff
@@ -1,26 +1,258 @@
+      const fqItem = frequencyList[i];
+      const statusDb = stdntIdVsStatusMap.get(fqItem.studentId);
+      if (statusDb===undefined || !statusDb == fqItem.generalAttendanceAllowed) {
+        return {isValid:false, problematicId: fqItem.studentId};
+      }
+    }
+    return {isValid:true, problematicId: null};
+  }
+  public async getAttendanceListByClassAndDate(
+    date: Date,
+    classId: number,
+  ): Promise<StudentListByClassAndDateResponseDTO> {
+    const attendanceList =
+      await this.frequencyQueryService.getStudentByClassAndDateAttendanceList(
+        classId,
+        date,
+      );
+    return {
+      classId: classId,
+      date: date.toISOString().split("T")[0],
+      studentList: attendanceList,
+    };
+  }
+  public async createAttendanceList(
+    date: Date,
+    classId: number,
+  ): Promise<StudentListByClassAndDateResponseDTO> {
+    const attendanceList = await this.frequencyRepository.getByClassIdAnDate(
+      classId,
+      date,
+    );
+    if (attendanceList.length > 0) {
+      throw new ConflictException(
+        `Class with ID ${classId} and date ${date.toISOString().split("T")[0]} already was created.`,
+      );
+    }
+    const enrolledStudents =
+      await this.frequencyQueryService.getStudentsByClassId(classId);
+    if (enrolledStudents.length === 0) {
+      throw new NotFoundException(
+        `The class with ID ${classId} has no enrolled students or wasn't found.`,
+      );
+    }
+    const frequencies: Frequency[] = [];
+    for (let index = 0; index < enrolledStudents.length; index++) {
+      const frequency = new Frequency({
+        id: null,
+        studentId: enrolledStudents[index].id,
+        classId: classId,
+        date: date,
+        status: FrequencyStatus.PRESENTE,
+        notes: null,
+      });
+      frequencies.push(frequency);
+    }
+    const newFrequenciesArray = frequencies.map(
+      (frequency) =>
+        new Frequency({
+          id: null,
+          studentId: frequency.getStudentId(),
+          classId: null,
+          date: frequency.getDate(),
+          status: FrequencyStatus.PRESENTE,
+          notes: null,
+        }),
+    );
+    const wasItDeleted =
+      await this.frequencyRepository.deleteManyByStudentAndClassAndDate(
+        newFrequenciesArray,
+      );
+    if (!wasItDeleted) {
+      throw new InternalServerErrorException(
+        "The system wasn't able to create a new class attendance",
+      );
+    }
+    const wasItCreated = await this.frequencyRepository.createMany(frequencies);
+    if (!wasItCreated) {
+      throw new InternalServerErrorException(
+        "The system wasn't able to create a new class attendance",
+      );
+    }
+    return await this.getAttendanceListByClassAndDate(date, classId);
+  }
+
+  public async updateAttendanceList(
+    updateDto: UpdateClassAttendanceRequestDTO,
+  ): Promise<void> {
+    const { classId, date, studentList } = updateDto;
```

Comment:
You could distructure the object directly in the props, without creating the consts, but the difference is negligible
```suggestion
    { classId, date, studentList }: UpdateClassAttendanceRequestDTO,
  ): Promise<void> {
```

Reply:
Vou manter como estava antes.